src/Button/Button.tsx

```diff
@@ -200,227 +197,317 @@ interface Props
   variant?: "fab";
 }
 
+/**
+ * all the types `as` can be
+ */
+type AsType = React.ReactElement | keyof JSX.IntrinsicElements;
+
+// Why isn't this in it's own file? I don't know how to make the default value
```

Comment:
comment still relevant?

Reply:
No, it's not. And because it's not relevant, I can move this to another file.